src/download.rs

```diff
@@ -34,7 +35,7 @@ struct LatestReleaseApiResponse {
 pub struct DownloadCfg {
     pub name: String,
     pub target: TargetTriple,
-    pub version: Version,
+    pub version: String,
```

Comment:
Hmm, on a second look I think your suggestion is be better. I missed out the fact that `semver` crate allows parsing of the prerelease identifier as well so `0.22.1-nightly` is a valid `Version`. 

However I'd like to tweak your suggestion a bit and do something like this:

```rust
struct DownloadCfg {
  pub version: Version, // parse either `0.21.2` or `0.21.2-nightly`
  pub date: Option<Date> // have an optional date field for nightlies
}

// Same for the Package struct which deserializes from a Channel
struct Package {
  pub target: HashMap<String, HashedBinary>,
  pub version: Version,
  pub date: Option<Date>
}
```

Would this be an OK solution you think? This way we don't have to maintain 2 different `Version`s.

Reply:
So I wrote the above without consideration the magic behind `serde`. Turns out I can do something like:

```rust
struct Version {
    core: Version,
    date: Option<Date>,
}
```
(note: core is something I got from the [semver spec](https://semver.org/#backusnaur-form-grammar-for-valid-semver-versions))

and write a custom deserialization function to create an optional Date. This makes it easier for `fuelup check` as well because we can simply decide whether to compare dates solely based on if it's `Some` (without needing to check the toolchain it's running on)